src/amm/uniswap_v2/factory.rs

```diff
@@ -17,80 +20,85 @@ use tracing::instrument;
+        function getPair(address tokenA, address tokenB) external view returns (address pair);
+        function allPairs(uint256 index) external view returns (address pair);
+        function allPairsLength() external view returns (uint256 length);
+    }
+}
 
-pub const PAIR_CREATED_EVENT_SIGNATURE: H256 = H256([
+pub const PAIR_CREATED_EVENT_SIGNATURE: B256 = FixedBytes([
     13, 54, 72, 189, 15, 107, 168, 1, 52, 163, 59, 169, 39, 90, 197, 133, 217, 211, 21, 240, 173,
     131, 85, 205, 222, 253, 227, 26, 250, 40, 208, 233,
 ]);
 
 #[derive(Default, Debug, Clone, Serialize, Deserialize)]
 pub struct UniswapV2Factory {
-    pub address: H160,
+    pub address: Address,
     pub creation_block: u64,
     pub fee: u32,
 }
 
 impl UniswapV2Factory {
-    pub fn new(address: H160, creation_block: u64, fee: u32) -> UniswapV2Factory {
+    pub fn new(address: Address, creation_block: u64, fee: u32) -> UniswapV2Factory {
         UniswapV2Factory {
             address,
             creation_block,
             fee,
         }
     }
 
-    pub async fn get_all_pairs_via_batched_calls<M: Middleware>(
+    pub async fn get_all_pairs_via_batched_calls<
+        T: Transport + Clone,
+        P: Provider<T, AnyNetwork>,
+    >(
         &self,
-        middleware: Arc<M>,
-    ) -> Result<Vec<AMM>, AMMError<M>> {
-        let factory = IUniswapV2Factory::new(self.address, middleware.clone());
+        provider: Arc<P>,
+    ) -> Result<Vec<AMM>, AMMError> {
+        let factory = IUniswapV2FactorySol::new(self.address, provider.clone());
 
-        let pairs_length: U256 = factory.all_pairs_length().call().await?;
+        let IUniswapV2FactorySol::allPairsLengthReturn {
+            length: pairs_length,
+        } = factory.allPairsLength().call().await?;
 
         let mut pairs = vec![];
-        let step = 766; //max batch size for this call until codesize is too large
-        let mut idx_from = U256::zero();
-        let mut idx_to = if step > pairs_length.as_usize() {
+        // NOTE: max batch size for this call until codesize is too large
+        let step = 766;
+        let mut idx_from = U256::ZERO;
+        let mut idx_to = if step > pairs_length.to::<usize>() {
             pairs_length
         } else {
             U256::from(step)
         };
 
-        for _ in (0..pairs_length.as_u128()).step_by(step) {
+        for _ in (0..pairs_length.to::<usize>()).step_by(step) {
             pairs.append(
                 &mut batch_request::get_pairs_batch_request(
                     self.address,
                     idx_from,
                     idx_to,
-                    middleware.clone(),
+                    provider.clone(),
                 )
                 .await?,
             );
 
             idx_from = idx_to;
 
-            if idx_to + step > pairs_length {
-                idx_to = pairs_length - 1
+            if idx_to + U256::from(step) > pairs_length {
+                idx_to = pairs_length - U256::from(1)
```

Comment:
Lets make `U256::from(1)` a const.

Reply:
U mean just making ```Rust const U256_ONE: U256 = U256::from(1);``` and re-using it everywhere where we need U256::from(1)? Since the ```idx_to``` cannot be const, afaik it wont make any diff at all, or am I wrong?

Reply:
We can update any instance of `U256::from(1) to be the following since `U256::from(1)` is not const. This will result in minor savings but is a low hanging change. Lmk if I am overlooking something here.

```
pub const U256_ONE: U256 = U256::from_limbs([1, 0, 0, 0]);
```

`U256::from()` actually is the following under the hood.

```rust
#[inline]
    #[must_use]
    #[track_caller]
    pub fn from<T>(value: T) -> Self
    where
        Self: UintTryFrom<T>,
    {
        match Self::uint_try_from(value) {
            Ok(n) => n,
            Err(e) => panic!("Uint conversion error: {e}"),
        }
    }
```

Reply:
nah nah, you are 100% correct, will do so, thx! P.S. I haven't looked into new alloy times under the hood yet, but it totally makes sense, thanks!